internal/layers/layers.go

```diff
@@ -285,6 +297,12 @@ func UnmountContainerLayers(ctx context.Context, layerFolders []string, containe
 	// Unload the SCSI scratch path
 	if (op & UnmountOperationSCSI) == UnmountOperationSCSI {
 		hostScratchFile := filepath.Join(layerFolders[len(layerFolders)-1], "sandbox.vhdx")
+		// Need to check if the sandbox is a symlink to another sandbox to handle the case
```

Comment:
Do we also need to update the `DestroyLayer` function call to _not_ delete the VHD if it is a symlink? I am not sure what the `vmcompute.DestroyLayer` API will do if the vhd is a symlink.

Reply:
DestroyLayer won't be called for LCOW layers which is the only time this would be a symlink so we're good here. We don't have any ideas for how we'd go about sharing scratch for WCOW yet.

Reply:
If we somehow arrived at a similar solution for WCOW, I don't believe we have to do anything special. DestroyLayer just calls `DeleteFile` on the vhd which seems like it handles symlinks just fine. https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-deletefile
```
Symbolic link behavior—

If the path points to a symbolic link, the symbolic link is deleted, not the target. To delete a target, you must call CreateFile and specify FILE_FLAG_DELETE_ON_CLOSE.
```

Reply:
Ah I am an idiot. I didn't realize that this was only for LCOW. Anyway we now also know that DestroyLayer will still work if we do this for WCOW.